### ISSUE TITLE
Subclass[BASE-STRING] must implement method[operator[]]

### ISSUE THREAD

**Issue by dg1sbg** [drmeister/clasp#314]
E.g. when doing:
```
> (ql:quickload "log4cl")
To load "log4cl":
  Load 1 ASDF system:
    alexandria
  Install 2 Quicklisp releases:
    bordeaux-threads log4cl
; Fetching #<URL "http://beta.quicklisp.org/archive/bordeaux-threads/2016-03-18/bordeaux-threads-v0.8.5.tgz">
; 19.63KB
==================================================
20,105 bytes in 0.02 seconds (1227.11KB/sec)
; Fetching #<URL "http://beta.quicklisp.org/archive/log4cl/2014-12-17/log4cl-20141217-git.tgz">
; 890.87KB
==================================================
912,253 bytes in 1.11 seconds (806.22KB/sec)
; Loading "log4cl"
[package bordeaux-threads]........................
[package log4cl-impl-fixer].......................
[package log4cl-impl].............

Condition of type: SIMPLE-PROGRAM-ERROR
**File(../../include/clasp/core/lispVector.h) lineNumber(61): Subclass[BASE-STRING] must implement method[operator[]]** 

Available restarts:
(use :r1 to invoke restart 1)

1. (RETRY) Retry compiling #<CL-SOURCE-FILE "log4cl" "src" "watcher">.
2. (ACCEPT) Continue, treating compiling #<CL-SOURCE-FILE "log4cl" "src" "watcher"> as having been successful.
3. (RETRY) Retry ASDF operation.
4. (CLEAR-CONFIGURATION-AND-RETRY) Retry ASDF operation after resetting the configuration.
5. (ABORT) Give up on "log4cl"
6. (RESTART-TOPLEVEL) Go back to Top-Level REPL.

Broken at frame[1405] CALL-WITH-VARIABLE-BOUND.
 File: #P"/opt/common-lisp/lang/clasp/src/clasp/src/core/compiler.cc" (Position #31803)
>> 
```

**Comment by Shinmera** [comment]
Can you narrow this down a little further than "log4cl"?

**Comment by dg1sbg** [comment]
When looking at the backtrace, these lines are causing the error:
1327 naming.lisp            393 (LOG4CL-IMPL::MAKE-LOG-EXPR-FORMAT ("Watcher thread already started") )
1326 naming.lisp              0 ('(:METHOD LOG4CL-IMPL:RESOLVE-LOGGER-FORM (T T T)) #<VA-LIST: (#<LOG4CL-IMPL> #<LEXICAL-VARIABLE LOG4CL-IMPL::LOGGER #<CLEAVIR-AST:LEXICAL-AST>> ("Watcher thread already started"))> 'NIL #<LOG4CL-IMPL> #<LEXICAL-VARIABLE LOG4CL-IMPL::LOGGER #<CLEAVIR-AST:LEXICAL-AST>> ("Watcher thread already started") )
1325 evaluator.cc            96 (APPLY #<CLOSURE-WITH-SLOTS '(:METHOD LOG4CL-IMPL:RESOLVE-LOGGER-FORM (T T T  )  )  :ftype :FUNCTION lambda-list: COMMON-LISP:NIL :fptr 0x115066fd0> #<VA-LIST: (#<LOG4CL-IMPL> #<LEXICAL-VARIABLE LOG4CL-IMPL::LOGGER #<CLEAVIR-AST:LEXICAL-AST>> ("Watcher thread alr)
1324 combin.lsp               0 (CLOS::COMBINE-METHOD-FUNCTIONS.LAMBDA #<VA-LIST: (#<LOG4CL-IMPL> #<LEXICAL-VARIABLE LOG4CL-IMPL::LOGGER #<CLEAVIR-AST:LEXICAL-AST>> ("Watcher thread already started"))> 'NIL )
1323 logger.lisp              0 (TOP-LEVEL )

**Comment by dg1sbg** [comment]
This bug may be closed as fixed.

**Comment by Shinmera** [comment]
Confirmed fixed on e3b1cc0d64f68d0853b2d375002a887afc4b6fdd.
